app/script/entity/User.js

```diff
@@ -247,27 +249,30 @@ z.entity.User = class User {
   }
 
   setRemaining(expirationTime) {
-    const timeToMinutes = 60 * 1000;
-    const timeToHours = timeToMinutes * 60;
+    const MILLISECONDS_IN_MINUTE = 60 * 1000;
+    const MILLISECONDS_IN_HOUR = MILLISECONDS_IN_MINUTE * 60;
 
     const remainingTime = Math.max(expirationTime - Date.now(), 0);
-    const remainingMinutes = Math.ceil(remainingTime / timeToMinutes);
+    const remainingMinutes = Math.ceil(remainingTime / MILLISECONDS_IN_MINUTE);
 
     let timeLeftText = z.string.userRemainingTimeHours;
     let timeValue = 0;
 
     if (remainingMinutes <= 60) {
       timeLeftText = z.string.userRemainingTimeMinutes;
       timeValue = Math.ceil(remainingMinutes / 15) * 15;
-      this.expirationRemaining(timeValue * timeToMinutes);
+      this.expirationRemaining(timeValue * MILLISECONDS_IN_MINUTE);
+      this.expirationRemainingText(`${timeValue}m`);
     } else if (remainingMinutes <= 90) {
       timeValue = 1.5;
-      this.expirationRemaining(timeValue * timeToHours);
+      this.expirationRemaining(timeValue * MILLISECONDS_IN_HOUR);
+      this.expirationRemainingText(`${timeValue}h`);
     } else {
       timeValue = Math.ceil(remainingMinutes / 60);
-      this.expirationRemaining(timeValue * timeToHours);
+      this.expirationRemaining(timeValue * MILLISECONDS_IN_HOUR);
+      this.expirationRemainingText(`${timeValue}h`);
     }
-
+    this.expirationIsUrgent(remainingMinutes < 120);
```

Comment:
If this is only used for the self user in the temporary guest left list we should consider moving it to the viewmodel and thereby removing it from all the user entities.